crates/types/src/consensus.rs

```diff
@@ -55,6 +55,9 @@ pub struct Consensus<TYPES: NodeType> {
     /// View number that is currently on.
     pub cur_view: TYPES::Time,
 
+    /// View we proposed in last.  To prevent duplicate proposals
+    pub last_proposed_view: TYPES::Time,
```

Comment:
How does this differ from `cur_view`?

Reply:
I guess the reason to separate `cur_view` and `last_propose_view` is that the former is for _any node_ to keep track of the view number it's on, whereas the latter is for a _leader_ to avoid duplicate proposals.

However, I don't think we are using `cur_view` except upon initialization (we are using the `cur_view` in each task instead), so we can probably just combine these two parameters.

Reply:
Yeah ideally `cur_view` dies sooner rather than later.